Source/WebCore/platform/graphics/avfoundation/objc/CDMSessionAVContentKeySession.mm

```diff
@@ -313,14 +313,22 @@ static bool isEqual(Uint8Array* data, const char* literal)
 
         errorCode = MediaPlayer::NoError;
         systemCode = 0;
-        NSError* error = nil;
         NSData* nsIdentifier = m_identifier ? [NSData dataWithBytes:m_identifier->data() length:m_identifier->length()] : contentKeyRequest.get().identifier;
 
-        NSData* requestData = [contentKeyRequest contentKeyRequestDataForApp:certificateData.get() contentIdentifier:nsIdentifier options:options.get() error:&error];
+        RetainPtr<NSError> error;
+        RetainPtr<NSData> requestData;
+        Semaphore keyRequestCompleted { 0 };
+        [contentKeyRequest makeStreamingContentKeyRequestDataForApp:certificateData.get() contentIdentifier:nsIdentifier options:options.get() completionHandler:[&] (NSData *outRequestData, NSError *outError) {
+            error = outError;
+            requestData = outRequestData;
+            keyRequestCompleted.signal();
+        }];
+        keyRequestCompleted.wait();
```

Comment:
Should this use `waitFor` so we can timeout, like we do when calling `processContentKeyRequestWithIdentifier`?

Reply:
I intentionally did not do that, because the block is referencing stack variables. If we timeout and continue, I suspect that the block will run after the stack has been torn down and cause a UAF.